sklearn/feature_extraction/text.py

```diff
@@ -1384,58 +1384,61 @@ class TfidfVectorizer(CountVectorizer):
         or more alphanumeric characters (punctuation is completely ignored
         and always treated as a token separator).
 
-    ngram_range : tuple (min_n, max_n)
+    ngram_range : tuple (min_n, max_n) (default=(1, 1))
         The lower and upper boundary of the range of n-values for different
         n-grams to be extracted. All values of n such that min_n <= n <= max_n
         will be used.
 
-    max_df : float in range [0.0, 1.0] or int, default=1.0
+    max_df : float in range [0.0, 1.0] or int (default=1.0)
         When building the vocabulary ignore terms that have a document
         frequency strictly higher than the given threshold (corpus-specific
         stop words).
         If float, the parameter represents a proportion of documents, integer
         absolute counts.
         This parameter is ignored if vocabulary is not None.
 
-    min_df : float in range [0.0, 1.0] or int, default=1
+    min_df : float in range [0.0, 1.0] or int (default=1)
         When building the vocabulary ignore terms that have a document
         frequency strictly lower than the given threshold. This value is also
         called cut-off in the literature.
         If float, the parameter represents a proportion of documents, integer
         absolute counts.
         This parameter is ignored if vocabulary is not None.
 
-    max_features : int or None, default=None
+    max_features : int or None (default=None)
         If not None, build a vocabulary that only consider the top
         max_features ordered by term frequency across the corpus.
 
         This parameter is ignored if vocabulary is not None.
 
-    vocabulary : Mapping or iterable, optional
+    vocabulary : Mapping or iterable, optional (default=None)
         Either a Mapping (e.g., a dict) where keys are terms and values are
         indices in the feature matrix, or an iterable over terms. If not
         given, a vocabulary is determined from the input documents.
 
-    binary : boolean, default=False
+    binary : boolean (default=False)
         If True, all non-zero term counts are set to 1. This does not mean
         outputs will have only 0/1 values, only that the tf term in tf-idf
         is binary. (Set idf and normalization to False to get 0/1 outputs.)
 
-    dtype : type, optional
+    dtype : type, optional (default=float64)
         Type of the matrix returned by fit_transform() or transform().
 
-    norm : 'l1', 'l2' or None, optional
-        Norm used to normalize term vectors. None for no normalization.
+    norm : 'l1', 'l2' or None, optional (default=’l2’)
+        Norm is used to normalize term vectors to have unit norm.
+        ``norm='l2'`` uses cosine similarity.
+        ``norm='l1'`` uses the Euclidean distance.
```

Comment:
This is interesting. Intuitively I wouldn't think that the Euclidean distance would result in an `l1` normalization. Is it really true?

Reply:
Maybe it should be: 
L2: Euclidean
L1:  Manhattan

Reply:
The norm is passed to the `normalize` function here: https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/preprocessing/data.py#L1518

That function also doesn't have a good explanation for the `norm` parameter. I suggest you figure out what exactly it does there, and fix the description in both places.

Reply:
Is this the level of detail that is needed:
```
The vector type to be used for normalization of the matrix is specified by the user (column or row).  The norms are computed as follows:

The L1 norm:
1.  converts each item in the vector to a positive value
2.  computes the total sum of these absolute values 
3.  divides each item in the vector by total sum of absolute values

The L2 norm:
1.  converts each item in the vector to its square
2.  computes the total sum of these squared values  
3.  divides each item in the vector by total sum of squared values
```

Reply:
I don't think that level of detail is appropriate.

l2 does imply cosine similarity when the dot product between vectors is taken, but that context is not mentioned.

l1 does not have anything to do with euclidean and does not result in a unit vector, so both statements are wrong. l1 might be used in text processing as a simple form of length normalisation, in the case that raw counts (not tf-idf) are used. Otherwise it is not commonly used in text processing.

I think referencing `normalize`/`Normalizer` and mentioning cosine similarity are appropriate. No motivation needs to be given for l1.

Reply:
How's this:
---
 norm : 'l1', 'l2' or None, optional (default=’l2’)
        The [normalize](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/preprocessing/data.py#L1518) function returns vectors in matrices with unit norm.  
        ``norm='l2'`` (for vectors normalized to unit length with only positive entries, this implies cosine similarity)
        ``norm='l1'``

Reply:
instead of `each output row`, should it be 
- `each output row/column` OR
- `each output vector`

Reply:
This parameter doesn't say anything about each column's distribution

Reply:
Ok, I will make the change in the text you included and exclude `:math:` since it is not rendering with `\log` or `\frac{}{}`.  Does that work?